src/main/java/lv/ctco/javaschool/goal/boundary/GoalApi.java

```diff
@@ -1,31 +1,146 @@
 package lv.ctco.javaschool.goal.boundary;
 
 import lv.ctco.javaschool.auth.control.UserStore;
+import lv.ctco.javaschool.auth.entity.domain.User;
 import lv.ctco.javaschool.goal.control.GoalStore;
+import lv.ctco.javaschool.goal.entity.Goal;
+import lv.ctco.javaschool.goal.entity.GoalDto;
+import lv.ctco.javaschool.goal.entity.GoalFormDto;
+import lv.ctco.javaschool.goal.entity.Tag;
 
 import javax.annotation.security.RolesAllowed;
-import javax.inject.Inject;
-import javax.persistence.EntityManager;
-import javax.persistence.PersistenceContext;
 import javax.ejb.Stateless;
-import javax.ws.rs.Path;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
 import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
+import static java.time.temporal.ChronoUnit.DAYS;
 
 @Path("/goal")
 @Stateless
 public class GoalApi {
-    @PersistenceContext
-    private EntityManager em;
     @Inject
     private UserStore userStore;
     @Inject
     private GoalStore goalStore;
 
+    @GET
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/mygoals")
+    public List<GoalDto> getMyGoals() {
+        User currentUser = userStore.getCurrentUser();
+        List<Goal> goalsList = goalStore.getGoalsListFor(currentUser);
+        if (goalsList.size() != 0) {
```

Comment:
`!goalsList.isEmpty()`

Reply:
done

Reply:
Do you need an explicit check for empty list? `goalsList.stream()...` will itself return an empty list if the input list contains no items.

Reply:
Check is not needed, method works well without this check, thanks!  :)